include/ISceneNode.h

```diff
@@ -276,30 +277,32 @@ namespace scene
 				child->grab();
 				child->remove(); // remove from old parent
 				Children.push_back(child);
+				// Note: This iterator is not invalidated until we erase it.
+				child->ThisIterator = Children.end();
```

Comment:
`child->ThisIterator = std::prev(Children.end());` or `child->ThisIterator = Children.insert(Children.end(), child);`

Reply:
Thanks for the suggestion!